src/coreclr/gcinfo/gcinfodumper.cpp

```diff
@@ -496,6 +506,9 @@ PORTABILITY_ASSERT("GcInfoDumper::ReportPointerRecord is not implemented on this
         pContext = (BYTE*)pRD->pCurrentContextPointers;
 #elif defined(TARGET_LOONGARCH64)
     assert(!"unimplemented on LOONGARCH yet");
+#elif defined(TARGET_RISCV64)
+    assert(!"unimplemented on RISCV64 yet");
+        pContext = (BYTE*)pRD->pCallerContext;
```

Comment:
Note, this is unimplemented block (and will not be implemented in this PR for sure). We stay with previous logic - assert in debug build and compile default code in release build. I can't remove any of this lines since this will broke previous logic - will not assert in debug or will not compile code in #else block.